pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/IntermediateResultsBlock.java

```diff
@@ -587,4 +600,101 @@ public BlockDocIdValueSet getBlockDocIdValueSet() {
   public BlockMetadata getMetadata() {
     throw new UnsupportedOperationException();
   }
+
+  public static class ListSelection extends IntermediateResultsBlock {
```

Comment:
Suggest not creating subclasses for now. For the purpose of this PR, we can add the `_comparator` to the `IntermediateResultsBlock` and change the constructor to `public IntermediateResultsBlock(DataSchema dataSchema, Collection<Object[]> selectionResult, @Nullable Comparator comparator, boolean nullHandlingEnabled)`

I'm planning to split the `IntermediateResultsBlock` into results block for each query type (similar idea with this one), but IMO it is better to do it in a separate PR

Reply:
I still don't like this solution because we don't have any type safety when ordered results are required. To be honest, I don't even like that the comparator is stored in this class instead of being part of the combinator. But I'm going to just add the comparator and pray for correctness in order to have this merged sooner.

Reply:
I agree. Ideally the comparator should be within the combine operator instead of the results block, and that's why I suggest not changing this class too much, but do a cleanup separately to move the comparator into the operator.